agent/statechange/statechange.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2014-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package statechange
+
+const (
+	// ContainerEvent is used to define the container state transition events
+	// emitted by the engine
+	ContainerEvent Event = iota
+
+	// TaskEvent is used to define the task state transition events emitted by
+	// the engine
+	TaskEvent
+)
+
+// Event defines the type of state change event
+type Event int32
```

Comment:
I'd initially thought of suggesting `EventType`, but didn't because `type EventType` was stutter-y. `EventKind` might be more appropriate. But, I don't have strong inclination towards any of these.